### PR TITLE
Add graphviz as submodule

### PR DESCRIPTION
**Pull request by dantti**

### CODE REVIEW

File: src/core/CMakeLists.txt [hunk 1]
```diff
@@ -117,7 +117,7 @@ if(GRAPHVIZ_FOUND)
     endif()
     target_link_libraries(kdstatemachineeditor_core LINK_PRIVATE ${GRAPHVIZ_GVC_LIBRARY} ${GRAPHVIZ_CGRAPH_LIBRARY})
     if(WITH_INTERNAL_GRAPHVIZ)
```
Comment (winterz) [inline src/core/CMakeLists.txt:119]: we don't need this conditional block now

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#
+# SPDX-License-Identifier: LGPL-2.1-only OR LicenseRef-KDAB-KDStateMachineEditor
+#
+# Licensees holding valid commercial KDAB State Machine Editor Library
+# licenses may use this file in accordance with the KDAB State Machine Editor
+# Library License Agreement provided with the Software.
+#
+# Contact info@kdab.com if any conditions of this licensing are not clear to you.
+#
+
```
Comment (winterz) [inline .gitmodules:15]: remove all that header stuff and add .gitmodules into .reuse/dep5

File: 3rdparty/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#
+# This file is part of the KDAB State Machine Editor Library.
+#
+# SPDX-FileCopyrightText: 2015-2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
```
Comment (winterz) [inline 3rdparty/CMakeLists.txt:4]: change "2015-2023" to "2024"

File: 3rdparty/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# Licensees holding valid commercial KDAB State Machine Editor Library
+# licenses may use this file in accordance with the KDAB State Machine Editor
+# Library License Agreement provided with the Software.
+#
+# Contact info@kdab.com if any conditions of this licensing are not clear to you.
+#
+
+if(WITH_INTERNAL_GRAPHVIZ)
+    # Function creates extra scope to keep these variables local
+    function(add_sonnet_subdirectory)
```
Comment (winterz) [inline 3rdparty/CMakeLists.txt:18]: sonnet?

**Review by winterz — CHANGES_REQUESTED** [review]

File: 3rdparty/CMakeLists.txt (hunk 4 above)
Comment (dantti) [reply 3rdparty/CMakeLists.txt:18]: weird that CI was happy

**Review by winterz — APPROVED** [review]

**Pull request merged**